docs/running-calyx/fud2/scripts.md

```diff
@@ -163,56 +163,4 @@ print(c::calyx_setup);
 
-</div>
-
-## API
-
-### High Level Rhai
-
-#### defop
-
-```
-defop <op name>(<input1>: <input1 state>, <input2>: <input2 state> ...) >> <target1>: <target1 state>, <target2>, <target2 state> ... {
-    <statements>
-}
-```
-Defines an op with `<op name>` which runs `<statements>` to generate target states from input states. 
-
-The name, inputs, and targets are called the signature of the op. `<statements>` is called the body of the op.
-
-#### shell
-
-```
-shell(<string>)
-```
-When called in the body of an op, that op will run `<string>` as a shell command to generate its targets. It is an error to call `shell` outside of the body of an op. Additionally, it is an error to call `shell` in the body of an op in which `shell_deps` was called prior.
-
-In the generated Ninja code, `shell` will create both a `rule` wrapping the shell command and a `build` command that invokes that rule. When a `defop` contains multiple `shell` commands, `fud2` automatically generates Ninja dependencies among the `build` command to ensure they run in order.
-
-#### shell_deps
-
-```
-shell_deps(<string>, [<dep1>, <dep2>, ...], [<target1>, <target2>, ..])
-```
-When called in the body of an op, that op will run `<string>` as a shell command if it needs to generate `<target1>, <target2>, ...` from `<dep1>, <dep2>, ...`. It is an error to call `shell_deps` outside of the body of an op. Additionally, it is an error to call `shell_deps` in the body of an op in which `shell`  was called prior.
-
-Targets and deps are either strings, such as `"file1"`, or identifiers, such as if `c: calyx` existed in the signature of an op then `c` could be used as a target or dep.
-
-A call to `shell_deps` corresponds directly to a Ninja rule in the Ninja file generated by `fud2`.
-
-### Low Level Rhai
-
-Currently, the Rhai API is almost identical to the Rust API. However `Emitter::add_file` is not currently supported. And `Emitter::var` is renamed to `_var` because `var` is a reserved keyword in Rhai.
-
-### Adding to the API
-
-If there is something that is hard to do in Rhai, it is straightforward to [register a Rust function][rhai-rust-fn] so that it is available from Rhai.
-
-Rust functions are registered in [`ScriptRunner::new`][fud-core-scriptrunner]. Refer to [`ScriptRunner::reg_get_state`][fud-core-reg_get_state] to see a simple example of how to register a function.
-
-[rhai]: https://rhai.rs/book/index.html
-[rhai-strings]: https://rhai.rs/book/ref/string-fn.html?highlight=String#standard-string-functions
-[rhai-rust-fn]: https://rhai.rs/book/rust/functions.html
-[fud2-scripts]: https://github.com/calyxir/calyx/tree/main/fud2/scripts
-[fud-core-scriptrunner]: https://github.com/calyxir/calyx/blob/6f895a1353020ce254860c3aa0fcfa2ba1abf4c4/fud2/fud-core/src/script/plugin.rs#L68
-[fud-core-reg_get_state]: https://github.com/calyxir/calyx/blob/6f895a1353020ce254860c3aa0fcfa2ba1abf4c4/fud2/fud-core/src/script/plugin.rs#L152
+</div>
```

Comment:
This page should probably link to the new Rhai page at some point.

Reply:
I made the new `rhai-api.md` supercede this page, and removed it. I hope this is ok?